src/upgrade.cpp

```diff
@@ -175,6 +175,16 @@ ConfigObject<ConfigValue>* Upgrade::versionUpgrade(const QString& settingsPath)
             delete oldFile;
```

Comment:
looks like oldFile is leaked here (not by your change) -- could you fix that? maybe just wrap it in a QScopedPointer so you don't have to delete in all the code paths.